### ISSUE TITLE
Check on debug launch configuration's `python` field prevents container debugging

### ISSUE THREAD

**Issue by bwateratmsft** [microsoft/debugpy#1517]
The Docker extension resolves a Python debug configuration in order to support Python debugging. Included in that resolved debug configuration is the `python` field, with a value of just `python3`. This is because this is the command to use for launching Python _in the container_. However, validation of that field is done against the host, so if the host and container don't both have `python3`, debugging fails. We've tried changing it to `python`, which worked on Windows, but failed on Mac and Linux.

Is there a way to override this check to stop it from validating against the host when it should not be?

An additional strange thing we've noticed is that the error seems to only show up on Insiders. The code doesn't seem to be Insiders only so it's confusing why it would be doing this.

**Comment by bwateratmsft** [comment]
It seems like it has started occurring in non-Insiders builds as well now.

**Comment by bwateratmsft** [comment]
> Thanks for that.
> 
> We do the validation yes, but I think this question is better suited for debugpy. We assume that the "python" specified contains a valid python binary in the host, but seems like that depends on the way launch config is specified?
> 
> If so, is there a way we can detect that, and thereby not do host validation in that case?

Perhaps if a custom `debugLauncherPath` is specified, the check should be skipped?

**Comment by karrtikr** [comment]
I think that could work, we should probably avoid adding a new flag though, unless absolutely needed.

**Comment by bwateratmsft** [comment]
What's confusing to me is that this is just now appearing, when a few days ago it was only showing up in Insiders, and before that not at all. It seems like this should have come up years ago.

**Comment by int19h** [comment]
debugpy itself doesn't do any validation in this case, it just uses the provided "python" etc values as is when building the command line. The validation code probably lives in https://github.com/microsoft/vscode-python-debugger?

**Comment by bwateratmsft** [comment]
We're not yet using the Python Debugger extension, blocked by https://github.com/microsoft/vscode-python-debugger/issues/194. We're still using the regular Python extension, so the validation is happening there.

**Comment by karrtikr** [comment]
@int19h Python extension does the validation, yes, but we do the validation assuming that "python" carries a valid executable in the host, which is not the case here as it refers to the "python" in the container instead.

What I wanted to check with debugpy is if there is a flag, like `debugLauncherPath` that we can check to approximately tell us that this "python" may not exist in the host, in which case we'll skip validation.

**Comment by int19h** [comment]
@karrtikr debugpy doesn't have any knowledge of that either. "debugLauncherPath" just tells it to use a custom launcher script, but that can be done for various reasons; using a wrapper script to effectively "remote" the launch, like Docker does, is only one possibility. I think that if some way to signal this is desirable, the extension should just introduce its own flag on debug config (which debugpy will simply ignore, as any other unknown property), and require clients that want to skip verification to set this flag also. 

However, I'm not sure that verification is a good idea here in the first place. Generally speaking, specifying "python" explicitly is already an advanced scenario (since the usual flow is to just select the appropriate interpreter from the status bar). Also, given that the launch command usually executes in a shell, even locally there's a legitimate case when "python" wouldn't be a valid executable binary: it can also be a shell alias. So e.g. this is valid:
```json5
"python": "exec"
"pythonArgs": ["custom-python-wrapper.sh"]
```
but I don't know how this can even be validated without actually running it in the user's choice of shell. 

By the way, what does the validator do when "python" is an array value? e.g. the above can also be written thus:
```json5
"python": ["exec", "custom-python-wrapper.sh"]
```

**Comment by karrtikr** [comment]
> Also, given that the launch command usually executes in a shell, even locally there's a legitimate case when "python" wouldn't be a valid executable binary: it can also be a shell alias. So e.g. this is valid:

Right. We run the interpreter specified as part of validation, so this scenario is covered.

> By the way, what does the validator do when "python" is an array value? e.g. the above can also be written thus:

`package.json` contribution currently states that this value can only be a string, so we need not to do this yet.
